### PR TITLE
Add delete product prompt

### PR DESCRIPTION
**Pull request by mehul0810**
## Description
This PR is for #16676 

## How Has This Been Implemented?
I have added a confirmation popup on `Delete Permanently` link under Trash Status of Products list in admin. When you will delete a product, it first goes into `Trash` and then deleting from Trash needs a confirmation popup and alert user.

**Video:** http://recordit.co/NQZn3aQicB

## Types of changes
New feature (non-breaking change which adds functionality)

### CODE REVIEW

**Comment by seb86** [comment]
I thought the confirmation was suppose to be for active products not just when you permantly delete it?

I think we need two confirmations. The first confirmation only to show if the product has made any sales and then the second confirmation that you have already done.

#### Confirmation Message
> This product has produced sales. Deleting the product may cause issues with your orders. Are you sure you want to delete?"

Let me know what you think.

**Comment by mehul0810** [comment]
@seb86 I have not added confirmation for initial deletion of product to `trash` as i think that if user deletes a product and moves it to trash then user can restore/undo the action performed which is one of the type of confirmation only. Let me know your thoughts.

**Comment by seb86** [comment]
I have given my thoughts.

**Comment by mikejolley** [comment]
I actually think this should be for the trash action rather than the perma option - simply because once trashed, things like orders will no longer link correctly, and items can sit in trash and auto-delete themselves after a period of time.

**Comment by seb86** [comment]
Perfect. Good job @mehul0810

**Comment by mehul0810** [comment]
@seb86 @mikejolley I have updated the PR based on the comments provided. Please check and let me know your thoughts.

File: assets/js/admin/quick-edit.js [hunk 1]
```diff
@@ -118,4 +118,13 @@ jQuery(function( $ ) {
 		}
 
 	});
+
+	$( '#wpbody' ).on( 'click', '.trash-product', function() {
+		var confirm_delete = confirm( woocommerce_admin.delete_product_message );
+
+		// Stop redirection, if user denies to delete the product.
+		if( ! confirm_delete ) {
```
Comment (mikejolley) [inline assets/js/admin/quick-edit.js]: Needs space before parentheses.

File: includes/admin/class-wc-admin-assets.php [hunk 2]
```diff
@@ -154,6 +154,7 @@ public function admin_scripts() {
 				'i18n_mon_decimal_error'            => sprintf( __( 'Please enter in monetary decimal (%s) format without thousand separators and currency symbols.', 'woocommerce' ), wc_get_price_decimal_separator() ),
 				'i18n_country_iso_error'            => __( 'Please enter in country code with two capital letters.', 'woocommerce' ),
 				'i18_sale_less_than_regular_error'  => __( 'Please enter in a value less than the regular price.', 'woocommerce' ),
+				'delete_product_message'            => __( 'This product has produced sales. Deleting the product may cause issues with your orders. Are you sure you want to delete?', 'woocommerce' ),
```
Comment (mikejolley) [inline includes/admin/class-wc-admin-assets.php]: > This product has produced sales and may be linked to existing orders. Are you sure you want to delete it?

File: includes/admin/class-wc-admin-duplicate-product.php [hunk 3]
```diff
@@ -43,6 +43,18 @@ public function dupe_link( $actions, $post ) {
 			return $actions;
 		}
 
+		// Add Class to Delete Permanently link in row actions.
+		$total_sales = get_post_meta( $post->ID, 'total_sales', true );
```
Comment (mikejolley) [inline includes/admin/class-wc-admin-duplicate-product.php]: This should be using CRUD, `get_total_sales` from a product object.

File: includes/admin/class-wc-admin-duplicate-product.php [hunk 4]
```diff
@@ -43,6 +43,18 @@ public function dupe_link( $actions, $post ) {
...
+		if( 'publish' === $post->post_status && 0 < $total_sales ) {
```
Comment (mikejolley) [inline includes/admin/class-wc-admin-duplicate-product.php]: Why is trash disabled if there are no sales, rather than giving the choice?

**Review by mikejolley — CHANGES_REQUESTED** [review]

File: includes/admin/class-wc-admin-duplicate-product.php (hunk 4 above)
Comment (mehul0810) [reply includes/admin/class-wc-admin-duplicate-product.php]: @mikejolley I've not disabled Trash when there is no sales. Trash will work as it is but the confirmation popup will not appear if there is no sales. Let me know if there is any change to the way of implementation.

File: includes/admin/class-wc-admin-duplicate-product.php (hunk 3 above)
Comment (mehul0810) [reply includes/admin/class-wc-admin-duplicate-product.php]: Implemented Change

File: includes/admin/class-wc-admin-assets.php (hunk 2 above)
Comment (mehul0810) [reply includes/admin/class-wc-admin-assets.php]: Implemented Change

File: assets/js/admin/quick-edit.js (hunk 1 above)
Comment (mehul0810) [reply assets/js/admin/quick-edit.js]: Implemented Change

File: assets/js/admin/quick-edit.js [hunk 5]
```diff
@@ -118,4 +118,13 @@ jQuery(function( $ ) {
 		}
 
 	});
+
+	$( '#wpbody' ).on( 'click', '.trash-product', function() {
+		var confirm_delete = confirm( woocommerce_admin.delete_product_message );
```
Comment (claudiosanches) [inline assets/js/admin/quick-edit.js]: Need to change `confirm` to `window.confirm`.

Comment (claudiosanches) [reply assets/js/admin/quick-edit.js]: In fact, you can change all this code to:

```js
$( '#wpbody' ).on( 'click', '.trash-product', function() {
	return window.confirm( woocommerce_admin.delete_product_message );
});
```

**Review by claudiosanches — CHANGES_REQUESTED** [review]
Missing this changes.

File: assets/js/admin/quick-edit.js (hunk 5 above)
Comment (mehul0810) [reply assets/js/admin/quick-edit.js]: implemented both the suggested changes

File: includes/admin/class-wc-admin-assets.php [hunk 6]
```diff
@@ -154,6 +154,7 @@ public function admin_scripts() {
 				'i18n_mon_decimal_error'            => sprintf( __( 'Please enter in monetary decimal (%s) format without thousand separators and currency symbols.', 'woocommerce' ), wc_get_price_decimal_separator() ),
 				'i18n_country_iso_error'            => __( 'Please enter in country code with two capital letters.', 'woocommerce' ),
 				'i18_sale_less_than_regular_error'  => __( 'Please enter in a value less than the regular price.', 'woocommerce' ),
+				'delete_product_message'            => __( 'This product has produced sales and may be linked to existing orders. Are you sure you want to delete it?', 'woocommerce' ),
```
[the remark this answers is not in the document]
Comment (mehul0810) [reply includes/admin/class-wc-admin-assets.php]: Implemented Change

File: includes/admin/class-wc-admin-duplicate-product.php [hunk 7]
```diff
@@ -43,6 +43,18 @@ public function dupe_link( $actions, $post ) {
...
 		}
 
+		// Add Class to Delete Permanently link in row actions.
+		$product = new WC_Product( $post->ID );
+		if( 'publish' === $post->post_status && 0 < $product->get_total_sales() ) {
+			$actions['trash'] = sprintf(
+				'<a href="%s" class="submitdelete trash-product" aria-label="%s">%s</a>',
+				get_delete_post_link( $post->ID, '', false ),
+				/* translators: %s: post title */
+				esc_attr( sprintf( __( 'Move &#8220;%s&#8221; to the Trash' ), $post->post_title ) ),
```
[the remark this answers is not in the document]
Comment (mehul0810) [reply includes/admin/class-wc-admin-duplicate-product.php]: Added Text Domain

File: includes/admin/class-wc-admin-duplicate-product.php [hunk 8]
```diff
@@ -43,6 +43,18 @@ public function dupe_link( $actions, $post ) {
...
+				__( 'Trash' )
```
[the remark this answers is not in the document]
Comment (mehul0810) [reply includes/admin/class-wc-admin-duplicate-product.php]: Added Text Domain

**Comment by mehul0810** [comment]
@mikejolley @claudiosanches I've implemented the required changes to the PR. Can you review and let me know if there are any further changes.

**Comment by myvcc657** [comment]
`perbaiki semua masalah`

**Comment by mikejolley** [comment]
Looks good now. Just one minor thing I'm correcting before merging; I think we can use a global for the product data that exists already. This will speed it up rather than loading a new product object. 👍

**Pull request merged**